### PR TITLE
NAP interface implementation - part 1

### PR DESCRIPTION
**Pull request by mwielgus**
* compilation fixes in kubemark
* autoprovisioned() bool function in NodeGroup
* minor refactoring in GCE cloud provider

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/cloud_provider.go [hunk 1]
```diff
@@ -119,6 +119,9 @@ type NodeGroup interface {
 	// This will be executed only for autoprovisioned node groups, once their size drops to 0.
 	// Implementation optional.
 	Delete() error
+
+	// Autoprovisioned returns true if the node group is autoprovisioned.
```
Comment (wasylkowski) [inline cluster-autoscaler/cloudprovider/cloud_provider.go]: What does it technically mean that a node group is autoprovisioned? Can you be more explicit here?

Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/cloud_provider.go]: Done.

File: cluster-autoscaler/cloudprovider/gce/gce_manager.go [hunk 2]
```diff
@@ -59,6 +61,8 @@ type GceManager struct {
 
 	service    *gce.Service
 	cacheMutex sync.Mutex
+	zone       string
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/gce_manager.go]: What about multi-zone clusters?

Comment (mwielgus) [reply cluster-autoscaler/cloudprovider/gce/gce_manager.go]: They won't be fully supported in the first version. NAP won't create regional multiple node pools.

**Comment by mwielgus** [comment]
ref: #225

**Pull request merged**